snapcraft/tests/plugins/test_python.py

```diff
@@ -121,26 +132,33 @@ def test_env(self):
         env_missing_path = plugin.env('/testpath')
         self.assertTrue('PYTHONPATH=/testpath' not in env_missing_path)
 
+    @mock.patch.object(python.PythonPlugin, 'run_output')
     @mock.patch.object(python.PythonPlugin, 'run')
     @mock.patch.object(os.path, 'exists', return_value=False)
-    def test_missing_setup_path(self, mock_path_exists, mock_run):
+    def test_missing_setup_path(
+            self, mock_path_exists, mock_run, mock_run_output):
+        mock_run_output.side_effect = fake_empty_pip_list
         plugin = python.PythonPlugin('test-part', self.options,
                                      self.project_options)
         setup_directories(plugin, self.options.python_version)
         plugin.pull()
         self.assertFalse(mock_run.called)
 
+    @mock.patch.object(python.PythonPlugin, 'run_output')
     @mock.patch.object(python.PythonPlugin, 'run')
-    def test_pull_with_nothing(self, mock_run):
+    def test_pull_with_nothing(self, mock_run, mock_run_output):
+        mock_run_output.side_effect = fake_empty_pip_list
```

Comment:
This can/should be done in the `@mock.patch.object` line - e.g. `@mock.patch.object(python.PythonPlugin, 'run_output', side_effect=fake_empty_pip_list)`

Makes a clearer distinction between your fixture and the actual test